### PR TITLE
Add CPP solution to usaco-832

### PR DESCRIPTION
**Pull request by alireza-jahandoost**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
tabs

File: solutions/usaco-832.mdx [hunk 1]
```diff
@@ -22,6 +22,95 @@ In this problem, there are three cases that we have to consider:
...
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main(){
+	freopen("milkorder.in", "r", stdin);
+	freopen("milkorder.out", "w", stdout);
+	int n,m,k;
+	cin>>n>>m>>k;
```
Comment (SansPapyrus683) [inline solutions/usaco-832.mdx:35]: spaces between operators

File: solutions/usaco-832.mdx [hunk 2]
```diff
@@ -22,6 +22,96 @@ In this problem, there are three cases that we have to consider:
...
+
+		// Case 2
+	if(statuspos.count(1)){
+		int start = statuspos[1] - 1;
+		while(start >= 0){
+			if(exist.count(status[start])) break;
+			start --;
+		}
+		bool check = start == -1;
+		int diff = statuspos[1] - start;
```
Comment (SansPapyrus683) [inline solutions/usaco-832.mdx:67]: indentation here is really weird

**Review by SansPapyrus683 — COMMENTED** [review]
@envyaims your turn

File: solutions/usaco-832.mdx [hunk 3]
```diff
@@ -22,6 +22,106 @@ In this problem, there are three cases that we have to consider:
...
+	// Case 2
+	if (statuspos.count(1)) {
+		int start = statuspos[1] - 1;
+		while (start >= 0) {
+			if (exist.count(status[start])) {
+				break;
+			}
+			start--;
+		}
+		bool check = start == -1;
```
Comment (envyaims) [inline solutions/usaco-832.mdx:71]:
```suggestion
		bool check = (start == -1);
```

File: solutions/usaco-832.mdx [hunk 4]
```diff
@@ -22,6 +22,106 @@ In this problem, there are three cases that we have to consider:
...
+		return 0;
+	}
+
+	// Case 3
+	for (int i = 1; i <= n; i++) {
+		if (pos[i] == -1) {
+			continue;
+		}
+		if (statuspos.count(pos[i])) {
+			int a = statuspos[pos[i]] - 1;
```
Comment (envyaims) [inline solutions/usaco-832.mdx:95]: what does a and b represent?

File: solutions/usaco-832.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/usaco-832.mdx:71]: also use a better variable name than `check`

File: solutions/usaco-832.mdx [hunk 5]
```diff
@@ -22,6 +22,106 @@ In this problem, there are three cases that we have to consider:
...
+
+int main() {
+	freopen("milkorder.in", "r", stdin);
+	freopen("milkorder.out", "w", stdout);
+
+	int n, m, k;
+	cin >> n >> m >> k;
+
+	vector<int> pos(n + 1, -1);
+	map<int, int> statuspos;
```
Comment (envyaims) [inline solutions/usaco-832.mdx:39]: add some more comments! what does this store?

**Review by envyaims — CHANGES_REQUESTED** [review]
yo

File: solutions/usaco-832.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-832.mdx:71]: parenthesis doesn't really matter

Comment (envyaims) [reply solutions/usaco-832.mdx:71]: makes it clearer since there are a lot of equal signs

Comment (SansPapyrus683) [reply solutions/usaco-832.mdx:71]: ykw personal preference, op i'll leave it up to you

**Pull request closed without merging**